src/main/java/io/carbynestack/cli/client/castor/util/TupleFileParser.java

```diff
@@ -35,12 +37,36 @@ public static TupleChunk parse(TupleType tupleType, File tupleFile, UUID chunkId
 
     byte[] bytes;
     try (FileInputStream fileInputStream = new FileInputStream(tupleFile)) {
-      bytes = toByteArray(fileInputStream);
+      final byte[] fileBytes = toByteArray(fileInputStream);
+      bytes = removeTupleHeader(fileBytes);
     } catch (IOException e) {
       log.error(String.format("Cannot read file %s", tupleFile.getPath()));
       throw new RuntimeException();
     }
 
     return TupleChunk.of(tupleType.getTupleCls(), tupleType.getField(), chunkId, bytes);
   }
+
+  /**
+   * Removes the File Header that was introduced in MP-SPDZ v0.2.8
+   *
+   * @param tuplesWithHeader The file bytes that include the header
+   * @return The file bytes shifted by the header
+   */
+  private static byte[] removeTupleHeader(byte[] tuplesWithHeader) {
```

Comment:
For large files it could potentially be more efficient to skip the header before reading instead of reading the whole file first and return a copy of a subsequence.
```java
private static byte[] removeTupleHeader(FileInputStream tupleStreamWithHeader) throws IOException {
    byte[] headerLengthBytes = IOUtils.toByteArray(tupleStreamWithHeader, 8);
    int headerLength = (int)ByteBuffer.wrap(headerLengthBytes).order(ByteOrder.LITTLE_ENDIAN).getLong();
    tupleStreamWithHeader.skip(8+headerLength);
    return toByteArray(tupleStreamWithHeader);
}
```

and then replace line #40f
```java 
final byte[] fileBytes = toByteArray(fileInputStream);
bytes = removeTupleHeader(fileBytes);
```

with 

```java
bytes = removeTupleHeader(fileInputStream);
```

Reply:
fixed